docker/llm/README.md

```diff
@@ -229,102 +236,131 @@ docker run -itd \
+### 3. Serving with FastChat
 
-FastChat determines the Model adapter to use through path matching. Therefore, in order to load models using IPEX-LLM, you need to make some modifications to the model's name.
+To serve using the Web UI, you need three main components: web servers that interface with users, model workers that host one or more models, and a controller to coordinate the web server and model workers.
 
-A special case is `ChatGLM` models. For these models, you do not need to do any changes after downloading the model and the `IPEX-LLM` backend will be used automatically.
+- #### **Step 1: Launch the Controller**
+  ```bash
+  python3 -m fastchat.serve.controller &
+  ```
 
+  This controller manages the distributed workers.
 
-### Start the service
+- #### **Step 2: Launch the model worker(s)**
 
-#### Serving with Web UI
+  Using IPEX-LLM in FastChat does not impose any new limitations on model usage. Therefore, all Hugging Face Transformer models can be utilized in FastChat.
+  ```bash
+  source ipex-llm-init -t
 
-To serve using the Web UI, you need three main components: web servers that interface with users, model workers that host one or more models, and a controller to coordinate the web server and model workers.
+  # Available low_bit format including sym_int4, sym_int8, bf16 etc.
+  python3 -m ipex_llm.serving.fastchat.ipex_llm_worker --model-path path/to/vicuna-7b-v1.5 --low-bit "sym_int4" --trust-remote-code --device "cpu" &
+  ```
+  Wait until the process finishes loading the model and you see "Uvicorn running on ...". The model worker will register itself to the controller.
 
-##### Launch the Controller
-```bash
-python3 -m fastchat.serve.controller
-```
+- #### **Step 3: Launch Gradio web server or RESTful API server**
+  You can launch Gradio web server to serve your models using the web UI or launch RESTful API server to serve with cURL.
 
-This controller manages the distributed workers.
+  - **Option 1: Serving with Web UI**
+    ```bash
+    python3 -m fastchat.serve.gradio_web_server &
+    ```
+    This is the user interface that users will interact with.
 
-##### Launch the model worker(s)
-```bash
-python3 -m ipex_llm.serving.model_worker --model-path lmsys/vicuna-7b-v1.3 --device cpu
-```
-Wait until the process finishes loading the model and you see "Uvicorn running on ...". The model worker will register itself to the controller.
+    By following these steps, you will be able to serve your models using the web UI with `IPEX-LLM` as the backend. You can open your browser and chat with a model now.
 
-> To run model worker using Intel GPU, simply change the --device cpu option to --device xpu
+  - **Option 2: Serving with OpenAI-Compatible RESTful APIs**
 
-##### Launch the Gradio web server
+      Launch the RESTful API server
 
-```bash
-python3 -m fastchat.serve.gradio_web_server
-```
+      ```bash
+      python3 -m fastchat.serve.openai_api_server --host localhost --port 8000 &
```

Comment:
localhost to 0.0.0.0? and please test serving with ip address.

Reply:
tested and updated, serving with ip address works.